Queueing with IQS fails

### Problem description:

The 1st item is ok, from the 2nd in queue it fails to convert

### HandBrake version (e.g., 1.3.0):
Build:
Lastest snapshot of HB from the 12th of March, it seems that only in lastest build its fucked up
does not detect Intel Quick Sync from 2nd video in queue and fails

### Operating system and version (e.g., Ubuntu 19.10, macOS 10.15 Catalina, Windows 10 1909):
OS:
Win 10 1909

HandBrake Nightly 20210312190521-48617ff44-master (2021031201)
OS: Microsoft Windows NT 10.0.18363.0
CPU: Intel(R) Core(TM) i3-8130U CPU @ 2.20GHz
Ram: 8075 MB, 
GPU Information:
  Intel(R) UHD Graphics 620 - 27.20.100.9168
Screen: 1920x1080
Temp Dir: C:\Users\Gamer\AppData\Local\Temp\
Install Dir: C:\Users\Gamer\Desktop\HandBrake\
Data Dir: C:\Users\Gamer\AppData\Roaming\HandBrake\Nightly

-------------------------------------------


# Starting Encode ...
[11:55:25] base preset: 480p
[11:55:25] hb_init: starting libhb thread
qsv: incorrect qsv device index -1
[11:55:25] Starting work at: Wed Mar 17 11:55:25 2021
[11:55:25] 1 job(s) to process
[11:55:25] CPU: Intel(R) Core(TM) i3-8130U CPU @ 2.20GHz
[11:55:25]  - Intel microarchitecture Kaby Lake
[11:55:25]  - logical processor count: 4
[11:55:25] Intel Quick Sync Video support: no
[11:55:25] hb_scan: path=J:\test\movie2.mkv, title_index=1
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:323: failed opening UDF image J:\test\movie2.mkv
src/libbluray/disc/disc.c:424: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:424: error opening file BDMV\BACKUP\index.bdmv
src/libbluray/bluray.c:2585: nav_get_title_list(J:\test\movie2.mkv\) failed
[11:55:26] bd: not a bd - trying as a stream/file instead
libdvdnav: Using dvdnav version 6.0.1
libdvdread: Encrypted DVD support unavailable.
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed
libdvdread: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: failed to read VIDEO_TS.IFO
[11:55:26] dvd: not a dvd - trying as a stream/file instead
Input #0, matroska,webm, from 'J:\test\movie2.mkv':
  Metadata:
    title           : ppp
    encoder         : libebml v1.4.1 + libmatroska v1.6.2
    creation_time   : 2021-02-19T13:08:32.000000Z
  Duration: 00:18:22.96, start: 0.000000, bitrate: 2689 kb/s
    Stream #0:0: Video: h264 (High), yuv420p(tv, bt709, progressive), 1280x720 [SAR 1:1 DAR 16:9], 24 fps, 24 tbr, 1k tbn, 48 tbc (default)
    Metadata:
      BPS-eng         : 2495547
      DURATION-eng    : 00:18:22.959000000
      NUMBER_OF_FRAMES-eng: 26471
      NUMBER_OF_BYTES-eng: 344060791
      _STATISTICS_WRITING_APP-eng: mkvmerge v53.0.0 ('Fool's Gold') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-02-19 13:08:32
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
    Stream #0:1(pol): Audio: ac3, 48000 Hz, stereo, fltp, 192 kb/s (default)
    Metadata:
      BPS-eng         : 192000
      DURATION-eng    : 00:18:22.944000000
      NUMBER_OF_FRAMES-eng: 34467
      NUMBER_OF_BYTES-eng: 26470656
      _STATISTICS_WRITING_APP-eng: mkvmerge v53.0.0 ('Fool's Gold') 64-bit
      _STATISTICS_WRITING_DATE_UTC-eng: 2021-02-19 13:08:32
      _STATISTICS_TAGS-eng: BPS DURATION NUMBER_OF_FRAMES NUMBER_OF_BYTES
[11:55:26] scan: decoding previews for title 1
[11:55:26] scan: audio 0x1: ac3, rate=48000Hz, bitrate=192000 polski (AC3) (2.0 ch) (192 kbps)
[11:55:26] scan: 10 previews, 1280x720, 24.000 fps, autocrop = 0/0/0/0, aspect 16:9, PAR 1:1, color profile: 1-1-1
[11:55:26] scan: supported video decoders: avcodec qsv
[11:55:26] libhb: scan thread found 1 valid title(s)
[11:55:26] Starting Task: Encoding Pass
[11:55:26] MTFrame thread started for segment 1
[11:55:26] MTFrame thread started for segment 0
[11:55:26] MTFrame thread started for segment 3
[11:55:26] MTFrame thread started for segment 2
[11:55:26] work: only 1 chapter, disabling chapter markers
[11:55:26] job configuration:
[11:55:26]  * source
[11:55:26]    + J:\test\movie2.mkv
[11:55:26]    + title 1, chapter(s) 1 to 1
[11:55:26]    + container: matroska,webm
[11:55:26]    + data rate: 2689 kbps
[11:55:26]  * destination
[11:55:26]    + J:\Movie2-1.mkv
[11:55:26]    + container: Matroska (libavformat)
[11:55:26]  * video track
[11:55:26]    + decoder: h264 8-bit
[11:55:26]    + filters
[11:55:26]      + Framerate Shaper (mode=1)
[11:55:26]        + frame rate: 24.000 fps -> constant 24.000 fps
[11:55:26]      + Crop and Scale (width=720:height=406:crop-top=0:crop-bottom=0:crop-left=0:crop-right=0)
[11:55:26]        + source: 1280 * 720, crop (0/0/0/0): 1280 * 720, scale: 720 * 406
[11:55:26]      + Sharpen (lapsharp) (y-strength=0.2:y-kernel=isolap:cb-strength=0.12:cb-kernel=isolap)
[11:55:26]    + Output geometry
[11:55:26]      + storage dimensions: 720 x 406
[11:55:26]      + pixel aspect ratio: 406 : 405
[11:55:26]      + display dimensions: 721 x 406
[11:55:26]    + encoder: H.264 (Intel Media SDK)
[11:55:26]      + preset:  quality
[11:55:26]      + options: lowpower=1
[11:55:26]      + profile: auto
[11:55:26]      + level:   auto
[11:55:26]      + bitrate: 1400 kbps, pass: 0
[11:55:26]      + color profile: 1-1-1
[11:55:26] sync: expecting 26471 video frames
qsv: incorrect qsv device index -1
Last error repeated 31 times
hb_qsv_param_default_preset: invalid pointer(s) param=000001ec1a281888 videoParam=000001ec1a2819b0 info=0000000000000000 preset=000001ec1a360200
encqsvInit: hb_qsv_param_default_preset failed
Failure to initialise thread 'Quick Sync Video encoder (Intel Media SDK)'
[11:55:26] vfr: 0 frames output, 0 dropped and 0 duped for CFR/PFR
[11:55:26] vfr: lost time: 0 (0 frames)
[11:55:26] vfr: gained time: 0 (0 frames) (0 not accounted for)
[11:55:26] h264-decoder done: 0 frames, 0 decoder errors
[11:55:26] sync: got 0 frames, 26471 expected
[11:55:26] Finished work at: Wed Mar 17 11:55:26 2021
[11:55:26] libhb: work result = 3

# Job Failed to Initialise. Check log and input settings (3)

I was able to easily and reliably reproduce this issue on 3 different Intel machines (6th, 8th & 9th gen).

All I did was unpack the nightly .zip into a temp dir, enable "portable.ini", run Handbrake.exe, **ONLY** change log level to Extended, load any media file at all, select any QSV encoder at all, then hit "Preview" twice in a row.

You can rinse/repeat the "load any media file part" onwards after restarting Handbrake.

Here's a sample log from my 6th gen.  8th & 9th were the same.  Latest drivers were the same as 1 year old drivers.  Any media file was the same.   Any QSV encoder was the same:

```
HandBrake Nightly 20210312190521-48617ff44-master (2021031201)
OS: Microsoft Windows NT 10.0.19042.0
CPU: Intel(R) Core(TM) i3-6100 CPU @ 3.70GHz
Ram: 15296 MB, 
GPU Information:
  Intel(R) HD Graphics 530 - 27.20.100.9079
Screen: 1920x1080
Temp Dir: Q:\i386\Video\Handbrake\HandBrake\tmp\
Install Dir: Q:\i386\Video\Handbrake\HandBrake\
Data Dir: Q:\i386\Video\Handbrake\HandBrake\storage\HandBrake\Nightly

-------------------------------------------


# Starting Encode ...
[09:39:32] hb_init: starting libhb thread
[09:39:32] thread 4 started ("libhb")
[09:39:32] thread 5 started ("work")
[09:39:32] Starting work at: Sun Mar 21 09:39:32 2021
[09:39:32] 1 job(s) to process
[09:39:32] json job:
{
  "Audio": {
    "AudioList": [
      {
        "Bitrate": 160,
        "DRC": 0,
        "Encoder": "av_aac",
        "Gain": 0,
        "Mixdown": 4,
        "NormalizeMixLevel": false,
        "Samplerate": 0,
        "Track": 0,
        "DitherMethod": 0
      }
    ],
    "CopyMask": [
      "copy:aac",
      "copy:ac3",
      "copy:dtshd",
      "copy:dts",
      "copy:eac3",
      "copy:flac",
      "copy:mp3",
      "copy:truehd",
      "copy:mp2"
    ],
    "FallbackEncoder": "ac3"
  },
  "Destination": {
    "ChapterList": [
      {
        "Name": "Chapter 1"
      }
    ],
    "ChapterMarkers": true,
    "AlignAVStart": true,
    "File": "C:\\Users\\Media Server\\Videos\\Distributor Walt Disney Enhanced-Dweu-1_preview.m4v",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": false
    },
    "Mux": "av_mp4"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 4,
        "Settings": {
          "mode": "7"
        }
      },
      {
        "ID": 3,
        "Settings": {
          "block-height": "16",
          "block-thresh": "40",
          "block-width": "16",
          "filter-mode": "2",
          "mode": "3",
          "motion-thresh": "1",
          "spatial-metric": "2",
          "spatial-thresh": "1"
        }
      },
      {
        "ID": 12,
        "Settings": {
          "crop-bottom": "52",
          "crop-left": "0",
          "crop-right": "118",
          "crop-top": "2",
          "height": "522",
          "width": "602"
        }
      },
      {
        "ID": 6,
        "Settings": {
          "mode": "0"
        }
      }
    ]
  },
  "PAR": {
    "Num": 64,
    "Den": 45
  },
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "preview",
      "Start": 2,
      "End": 2700000,
      "SeekPoints": 10
    },
    "Title": 1,
    "Path": "V:\\0-9\\!!!\\S01\\distributor_walt_disney_enhanced-DWEU.vob"
  },
  "Subtitle": {
    "Search": {
      "Burn": false,
      "Default": false,
      "Enable": false,
      "Forced": false
    },
    "SubtitleList": []
  },
  "Video": {
    "Encoder": "qsv_h265",
    "Level": "auto",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "",
    "Preset": "balanced",
    "Profile": "auto",
    "Quality": 22,
    "QSV": {
      "Decode": true
    }
  }
}
[09:39:32] CPU: Intel(R) Core(TM) i3-6100 CPU @ 3.70GHz
[09:39:32]  - Intel microarchitecture Skylake
[09:39:32]  - logical processor count: 4
[09:39:32] Intel Quick Sync Video support: yes, gpu list: 0
[09:39:32] Intel Quick Sync Video integrated adapter with index 0
[09:39:32]  - Intel Media SDK hardware: API 1.34 (minimum: 1.3)
[09:39:32]  - Decode support:  h264 hevc (8bit: yes, 10bit: no)
[09:39:32]  - H.264 encoder: yes
[09:39:32]     - preferred implementation: hardware (1) via D3D11
[09:39:32]     - capabilities (hardware):  breftype icq+la+i+downs vsinfo opt1 opt2+mbbrc+extbrc+trellis+ib_adapt+nmpslice
[09:39:32]  - H.265 encoder: yes (8bit: yes, 10bit: no)
[09:39:32]     - preferred implementation: hardware (1) via D3D11
[09:39:32]     - capabilities (hardware):  bpyramid icq vsinfo opt1
[09:39:32] hb_scan: path=V:\0-9\!!!\S01\distributor_walt_disney_enhanced-DWEU.vob, title_index=1
[09:39:32] thread 6 started ("scan")
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:323: failed opening UDF image V:\0-9\!!!\S01\distributor_walt_disney_enhanced-DWEU.vob
src/libbluray/disc/disc.c:424: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:424: error opening file BDMV\BACKUP\index.bdmv
src/libbluray/bluray.c:2585: nav_get_title_list(V:\0-9\!!!\S01\distributor_walt_disney_enhanced-DWEU.vob\) failed
[09:39:32] bd: not a bd - trying as a stream/file instead
libdvdnav: Using dvdnav version 6.0.1
libdvdread: Encrypted DVD support unavailable.
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed
libdvdread: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: failed to read VIDEO_TS.IFO
[09:39:32] dvd: not a dvd - trying as a stream/file instead
[09:39:32] file is MPEG Program Stream
[09:39:32] Probing 1 unknown stream
[09:39:32]     Probe: Found stream mpegvideo. stream id 0xe0-0x0
[09:39:32] Found the following streams
[09:39:32]     Video Streams :
[09:39:32]       0xe0-0x0 type mpegvideo (0x2)
[09:39:32]     Audio Streams :
[09:39:32]       0xbd-0x80 type AC3 (0x81)
[09:39:32]     Subtitle Streams :
[09:39:32]     Other Streams :
[09:39:32] stream id 0xbd (type 0x81 substream 0x80) audio 0x8000bd
[09:39:32] hb_sample_pts: couldn't find video packet near 7081984
[09:39:32] hb_sample_pts: couldn't find video packet near 7139328
[09:39:32] hb_sample_pts: couldn't find video packet near 7198720
[09:39:32] hb_sample_pts: couldn't find video packet near 7256064
[09:39:32] hb_sample_pts: couldn't find video packet near 7313408
[09:39:32] hb_sample_pts: couldn't find video packet near 7370752
[09:39:32] scan: decoding previews for title 1
[09:39:32] file is MPEG Program Stream
[09:39:32] Probing 1 unknown stream
[09:39:32]     Probe: Found stream mpegvideo. stream id 0xe0-0x0
[09:39:32] scan: preview 1
[09:39:32] fifo_close: trashing 0 buffer(s)
[09:39:32] scan: audio 0x8000bd: ac3, rate=48000Hz, bitrate=448000 Unknown (AC3) (5.1 ch) (448 kbps)
[09:39:32] scan: preview 2
[09:39:32] scan: preview 3
[09:39:32] scan: preview 4
[09:39:32] scan: preview 5
[09:39:32] scan: preview 6
[09:39:32] scan: preview 7
[09:39:32] scan: preview 8
[09:39:32] scan: preview 9
[09:39:32] scan: preview 10
[09:39:32] stream: 73 good frames, 0 errors (0%)
[09:39:32] scan: 10 previews, 720x576, 25.000 fps, autocrop = 2/52/0/118, aspect 16:9, PAR 64:45, color profile: 5-1-6
[09:39:32] Freed 1 buffers of size 4096
[09:39:32] Freed 2 buffers of size 65536
[09:39:32] Freed 9 buffers of size 1048576
[09:39:32] Freed 1 buffers of size 2097152
[09:39:32] Allocated 11669504 bytes of buffers on this pass and Freed 11669504 bytes, 0 bytes leaked
[09:39:32] thread 6 exited ("scan")
[09:39:32] thread 6 joined ("scan")
[09:39:32] libhb: scan thread found 1 valid title(s)
[09:39:32] Starting Task: Encoding Pass
[09:39:32] Skipping vfr filter
[09:39:32] thread 7 started ("decomb_filter_segment")
[09:39:32] thread 8 started ("decomb_filter_segment")
[09:39:32] thread 9 started ("decomb_filter_segment")
[09:39:32] thread a started ("decomb_filter_segment")
[09:39:32] thread b started ("decomb_check_segment")
[09:39:32] thread c started ("decomb_check_segment")
[09:39:32] thread d started ("decomb_check_segment")
[09:39:32] thread e started ("decomb_check_segment")
[09:39:32] thread f started ("mask_filter_segment")
[09:39:32] thread 10 started ("mask_filter_segment")
[09:39:32] thread 11 started ("mask_filter_segment")
[09:39:32] thread 12 started ("mask_filter_segment")
[09:39:32] thread 13 started ("mask_erode_segment")
[09:39:32] thread 14 started ("mask_erode_segment")
[09:39:32] thread 15 started ("mask_erode_segment")
[09:39:32] thread 16 started ("mask_erode_segment")
[09:39:32] thread 17 started ("mask_dilate_segment")
[09:39:32] thread 18 started ("mask_dilate_segment")
[09:39:32] thread 19 started ("mask_dilate_segment")
[09:39:32] thread 1a started ("mask_dilate_segment")
[09:39:32] thread 1b started ("yadif_filter_segment")
[09:39:32] thread 1c started ("yadif_filter_segment")
[09:39:32] thread 1d started ("yadif_filter_segment")
[09:39:32] thread 1e started ("yadif_filter_segment")
[09:39:32] work: track 1, dithering not supported by codec
[09:39:32] work: only 1 chapter, disabling chapter markers
[09:39:32] job configuration:
[09:39:32]  * source
[09:39:32]    + V:\0-9\!!!\S01\distributor_walt_disney_enhanced-DWEU.vob
[09:39:32]    + title 1, start 00:00:0.00 stop 00:00:30.00
[09:39:32]  * destination
[09:39:32]    + C:\Users\Media Server\Videos\Distributor Walt Disney Enhanced-Dweu-1_preview.m4v
[09:39:32]    + container: MPEG-4 (libavformat)
[09:39:32]      + align initial A/V stream timestamps
[09:39:32]  * video track
[09:39:32]    + decoder: mpeg2video 8-bit
[09:39:32]      + bitrate 200 kbps
[09:39:32]    + filters
[09:39:32]      + Comb Detect (mode=3:spatial-metric=2:motion-thresh=1:spatial-thresh=1:filter-mode=2:block-thresh=40:block-width=16:block-height=16)
[09:39:32]      + Decomb (mode=39)
[09:39:32]      + AVFilter (crop='x=0:y=2:w=602:h=522',scale='width=602:height=522:flags=lanczos+accurate_rnd:in_color_matrix=smpte170m:out_color_matrix=smpte170m:out_range=limited',format='pix_fmts=yuv420p')
[09:39:32]        + crop: x=0, y=2, w=602, h=522
[09:39:32]        + scale: width=602, height=522, flags=lanczos+accurate_rnd, in_color_matrix=smpte170m, out_color_matrix=smpte170m, out_range=limited
[09:39:32]        + format: pix_fmts=yuv420p
[09:39:32]      + Crop and Scale (width=602:height=522:crop-top=2:crop-bottom=52:crop-left=0:crop-right=118)
[09:39:32]        + source: 720 * 576, crop (2/52/0/118): 602 * 522, scale: 602 * 522
[09:39:32]    + Output geometry
[09:39:32]      + storage dimensions: 602 x 522
[09:39:32]      + pixel aspect ratio: 64 : 45
[09:39:32]      + display dimensions: 856 x 522
[09:39:32]    + encoder: H.265 (Intel Media SDK)
[09:39:32]      + preset:  balanced
[09:39:32]      + profile: auto
[09:39:32]      + level:   auto
[09:39:32]      + quality: 22.00 (ICQ)
[09:39:32]      + color profile: 5-1-6
[09:39:32]  * audio track 1
[09:39:32]    + decoder: Unknown (AC3) (5.1 ch) (448 kbps) (track 1, id 0x8000bd)
[09:39:32]      + bitrate: 448 kbps, samplerate: 48000 Hz
[09:39:32]    + mixdown: Stereo
[09:39:32]    + encoder: AAC (libavcodec)
[09:39:32]      + bitrate: 160 kbps, samplerate: 48000 Hz
[09:39:32] file is MPEG Program Stream
[09:39:32] Probing 1 unknown stream
[09:39:32]     Probe: Found stream mpegvideo. stream id 0xe0-0x0
[09:39:32] sync: expecting 775 video frames
[09:39:32] thread 1f started ("Audio Synchronization")
[09:39:32] qsv: integrated qsv adapter with index 0 has been selected
[09:39:32] encqsvInit: using encode-only path
[09:39:32] encqsvInit: H.265/HEVC Main profile @ level 3.0
[09:39:32] encqsvInit: TargetUsage 4 AsyncDepth 4
[09:39:32] encqsvInit: GopRefDist 3 GopPicSize 25 NumRefFrame 2
[09:39:32] encqsvInit: BFramesMax 2 BRefType off
[09:39:32] encqsvInit: RateControlMethod ICQ ICQQuality 22
[09:39:32] Writing Metadata to output file...
[09:39:32] thread 20 started ("Muxer")
[09:39:32] thread 21 started ("Reader")
[09:39:32] thread 22 started ("Audio decoder (libavcodec)")
[09:39:32] thread 23 started ("Video decoder (libavcodec)")
[09:39:32] thread 24 started ("Video Synchronization")
[09:39:32] thread 25 started ("AVCodec Audio encoder (libavcodec)")
[09:39:32] thread 26 started ("Quick Sync Video encoder (Intel Media SDK)")
[09:39:32] thread 27 started ("Muxer")
[09:39:32] thread 28 started ("Comb Detect")
[09:39:32] thread 29 started ("Decomb")
[09:39:32] thread 2a started ("AVFilter")
[09:39:32] sync: first pts video is 0
[09:39:32] sync: first pts audio 0x8000bd is 0
[09:39:32] qsv_enc_init: using 'hardware (1) via D3D11' implementation, API: 1.34
[09:39:32] compute_init_delay: 3600 (1 frames)
[09:39:33] reader: done. 2 scr changes
[09:39:33] thread 21 exited ("Reader")
[09:39:33] 7.872000s: Video -> Film
[09:39:33] thread 27 exited ("Muxer")
[09:39:33] thread 27 joined ("Muxer")
[09:39:33] work: average encoding speed for job is 0.000000 fps
[09:39:33] thread 1f exited ("Audio Synchronization")
[09:39:33] thread 22 exited ("Audio decoder (libavcodec)")
[09:39:33] thread 23 exited ("Video decoder (libavcodec)")
[09:39:33] thread 28 exited ("Comb Detect")
[09:39:33] thread 28 joined ("Comb Detect")
[09:39:33] comb detect: heavy 0 | light 12 | uncombed 196 | total 208
[09:39:33] thread 7 exited ("decomb_filter_segment")
[09:39:33] thread 8 exited ("decomb_filter_segment")
[09:39:33] thread a exited ("decomb_filter_segment")
[09:39:33] thread 9 exited ("decomb_filter_segment")
[09:39:33] thread 7 joined ("decomb_filter_segment")
[09:39:33] thread 8 joined ("decomb_filter_segment")
[09:39:33] thread 9 joined ("decomb_filter_segment")
[09:39:33] thread a joined ("decomb_filter_segment")
[09:39:33] thread c exited ("decomb_check_segment")
[09:39:33] thread d exited ("decomb_check_segment")
[09:39:33] thread e exited ("decomb_check_segment")
[09:39:33] thread b exited ("decomb_check_segment")
[09:39:33] thread b joined ("decomb_check_segment")
[09:39:33] thread c joined ("decomb_check_segment")
[09:39:33] thread d joined ("decomb_check_segment")
[09:39:33] thread e joined ("decomb_check_segment")
[09:39:33] thread 11 exited ("mask_filter_segment")
[09:39:33] thread 12 exited ("mask_filter_segment")
[09:39:33] thread f exited ("mask_filter_segment")
[09:39:33] thread 10 exited ("mask_filter_segment")
[09:39:33] thread f joined ("mask_filter_segment")
[09:39:33] thread 10 joined ("mask_filter_segment")
[09:39:33] thread 11 joined ("mask_filter_segment")
[09:39:33] thread 12 joined ("mask_filter_segment")
[09:39:33] thread 16 exited ("mask_erode_segment")
[09:39:33] thread 15 exited ("mask_erode_segment")
[09:39:33] thread 14 exited ("mask_erode_segment")
[09:39:33] thread 13 exited ("mask_erode_segment")
[09:39:33] thread 13 joined ("mask_erode_segment")
[09:39:33] thread 14 joined ("mask_erode_segment")
[09:39:33] thread 15 joined ("mask_erode_segment")
[09:39:33] thread 16 joined ("mask_erode_segment")
[09:39:33] thread 18 exited ("mask_dilate_segment")
[09:39:33] thread 19 exited ("mask_dilate_segment")
[09:39:33] thread 1a exited ("mask_dilate_segment")
[09:39:33] thread 17 exited ("mask_dilate_segment")
[09:39:33] thread 17 joined ("mask_dilate_segment")
[09:39:33] thread 18 joined ("mask_dilate_segment")
[09:39:33] thread 19 joined ("mask_dilate_segment")
[09:39:33] thread 1a joined ("mask_dilate_segment")
[09:39:33] thread 24 exited ("Video Synchronization")
[09:39:34] thread 29 exited ("Decomb")
[09:39:34] thread 29 joined ("Decomb")
[09:39:34] decomb: deinterlaced 0 | blended 12 | unfiltered 196 | total 208
[09:39:34] thread 1b exited ("yadif_filter_segment")
[09:39:34] thread 1e exited ("yadif_filter_segment")
[09:39:34] thread 1c exited ("yadif_filter_segment")
[09:39:34] thread 1d exited ("yadif_filter_segment")
[09:39:34] thread 1b joined ("yadif_filter_segment")
[09:39:34] thread 1c joined ("yadif_filter_segment")
[09:39:34] thread 1d joined ("yadif_filter_segment")
[09:39:34] thread 1e joined ("yadif_filter_segment")
[09:39:34] thread 2a exited ("AVFilter")
[09:39:34] thread 2a joined ("AVFilter")
[09:39:34] thread 21 joined ("Reader")
[09:39:34] thread 22 joined ("Audio decoder (libavcodec)")
[09:39:34] thread 23 joined ("Video decoder (libavcodec)")
[09:39:34] thread 24 joined ("Video Synchronization")
[09:39:34] thread 20 exited ("Muxer")
[09:39:34] thread 25 exited ("AVCodec Audio encoder (libavcodec)")
[09:39:34] thread 25 joined ("AVCodec Audio encoder (libavcodec)")
[09:39:34] thread 26 exited ("Quick Sync Video encoder (Intel Media SDK)")
[09:39:34] thread 26 joined ("Quick Sync Video encoder (Intel Media SDK)")
[09:39:34] stream: 2902 good frames, 0 errors (0%)
[09:39:34] ac3-decoder done: 245 frames, 0 decoder errors
[09:39:34] mpeg2video-decoder done: 202 frames, 0 decoder errors
[09:39:34] sync: got 208 frames, 775 expected
[09:39:34] sync: framerate min 13.889 fps, max 25.000 fps, avg 24.904 fps
[09:39:34] thread 1f joined ("Audio Synchronization")
[09:39:34] encavcodecaudio: closing libavcodec
[aac @ 0000021d734d3dc0] Ignoring attempt to flush encoder that doesn't support it
[aac @ 0000021d734d3dc0] Qavg: 5086.291
[09:39:34] mux: file size, 1043541 bytes
[09:39:34] mux: track 0, 208 frames, 888286 bytes, 850.03 kbps, fifo 256
[09:39:34] mux: track 1, 368 frames, 147794 bytes, 141.43 kbps, fifo 512
[09:39:34] mux: overhead, 12.95 bytes per frame
[09:39:34] thread 20 joined ("Muxer")
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] fifo_close: trashing 0 buffer(s)
[09:39:34] Freed 320 buffers of size 1024
[09:39:34] Freed 25 buffers of size 2048
[09:39:34] Freed 32 buffers of size 4096
[09:39:34] Freed 32 buffers of size 8192
[09:39:34] Freed 32 buffers of size 16384
[09:39:34] Freed 5 buffers of size 32768
[09:39:34] Freed 1 buffers of size 65536
[09:39:34] Freed 7 buffers of size 524288
[09:39:34] Freed 32 buffers of size 1048576
[09:39:34] Allocated 38750208 bytes of buffers on this pass and Freed 38750208 bytes, 0 bytes leaked
[09:39:34] Finished work at: Sun Mar 21 09:39:34 2021
[09:39:34] thread 5 exited ("work")
[09:39:34] thread 5 joined ("work")
[09:39:34] libhb: work result = 0

# Job Completed!


***********************************************************************************************************************
[I TOOK 4 SECONDS TO PRESS "PREVIEW" AGAIN]
***********************************************************************************************************************


HandBrake Nightly 20210312190521-48617ff44-master (2021031201)
OS: Microsoft Windows NT 10.0.19042.0
CPU: Intel(R) Core(TM) i3-6100 CPU @ 3.70GHz
Ram: 15296 MB, 
GPU Information:
  Intel(R) HD Graphics 530 - 27.20.100.9079
Screen: 1920x1080
Temp Dir: Q:\i386\Video\Handbrake\HandBrake\tmp\
Install Dir: Q:\i386\Video\Handbrake\HandBrake\
Data Dir: Q:\i386\Video\Handbrake\HandBrake\storage\HandBrake\Nightly

-------------------------------------------


# Starting Encode ...
[09:39:38] thread 4 exited ("libhb")
[09:39:38] thread 4 joined ("libhb")
[09:39:38] hb_init: starting libhb thread
[09:39:38] thread 2c started ("libhb")
qsv: incorrect qsv device index -1
[09:39:38] thread 2d started ("work")
[09:39:38] Starting work at: Sun Mar 21 09:39:38 2021
[09:39:38] 1 job(s) to process
[09:39:38] json job:
{
  "Audio": {
    "AudioList": [
      {
        "Bitrate": 160,
        "DRC": 0,
        "Encoder": "av_aac",
        "Gain": 0,
        "Mixdown": 4,
        "NormalizeMixLevel": false,
        "Samplerate": 0,
        "Track": 0,
        "DitherMethod": 0
      }
    ],
    "CopyMask": [
      "copy:aac",
      "copy:ac3",
      "copy:dtshd",
      "copy:dts",
      "copy:eac3",
      "copy:flac",
      "copy:mp3",
      "copy:truehd",
      "copy:mp2"
    ],
    "FallbackEncoder": "ac3"
  },
  "Destination": {
    "ChapterList": [
      {
        "Name": "Chapter 1"
      }
    ],
    "ChapterMarkers": true,
    "AlignAVStart": true,
    "File": "C:\\Users\\Media Server\\Videos\\Distributor Walt Disney Enhanced-Dweu-1_preview.m4v",
    "Mp4Options": {
      "IpodAtom": false,
      "Mp4Optimize": false
    },
    "Mux": "av_mp4"
  },
  "Filters": {
    "FilterList": [
      {
        "ID": 4,
        "Settings": {
          "mode": "7"
        }
      },
      {
        "ID": 3,
        "Settings": {
          "block-height": "16",
          "block-thresh": "40",
          "block-width": "16",
          "filter-mode": "2",
          "mode": "3",
          "motion-thresh": "1",
          "spatial-metric": "2",
          "spatial-thresh": "1"
        }
      },
      {
        "ID": 12,
        "Settings": {
          "crop-bottom": "52",
          "crop-left": "0",
          "crop-right": "118",
          "crop-top": "2",
          "height": "522",
          "width": "602"
        }
      },
      {
        "ID": 6,
        "Settings": {
          "mode": "0"
        }
      }
    ]
  },
  "PAR": {
    "Num": 64,
    "Den": 45
  },
  "SequenceID": 0,
  "Source": {
    "Angle": 1,
    "Range": {
      "Type": "preview",
      "Start": 2,
      "End": 2700000,
      "SeekPoints": 10
    },
    "Title": 1,
    "Path": "V:\\0-9\\!!!\\S01\\distributor_walt_disney_enhanced-DWEU.vob"
  },
  "Subtitle": {
    "Search": {
      "Burn": false,
      "Default": false,
      "Enable": false,
      "Forced": false
    },
    "SubtitleList": []
  },
  "Video": {
    "Encoder": "qsv_h265",
    "Level": "auto",
    "TwoPass": false,
    "Turbo": false,
    "ColorMatrixCode": 0,
    "Options": "lowpower=1",
    "Preset": "balanced",
    "Profile": "auto",
    "Quality": 22,
    "QSV": {
      "Decode": false
    }
  }
}
[09:39:38] CPU: Intel(R) Core(TM) i3-6100 CPU @ 3.70GHz
[09:39:38]  - Intel microarchitecture Skylake
[09:39:38]  - logical processor count: 4
[09:39:38] Intel Quick Sync Video support: no
[09:39:38] hb_scan: path=V:\0-9\!!!\S01\distributor_walt_disney_enhanced-DWEU.vob, title_index=1
[09:39:38] thread 2e started ("scan")
udfread ERROR: ECMA 167 Volume Recognition failed
src/libbluray/disc/disc.c:323: failed opening UDF image V:\0-9\!!!\S01\distributor_walt_disney_enhanced-DWEU.vob
src/libbluray/disc/disc.c:424: error opening file BDMV\index.bdmv
src/libbluray/disc/disc.c:424: error opening file BDMV\BACKUP\index.bdmv
src/libbluray/bluray.c:2585: nav_get_title_list(V:\0-9\!!!\S01\distributor_walt_disney_enhanced-DWEU.vob\) failed
[09:39:38] bd: not a bd - trying as a stream/file instead
libdvdnav: Using dvdnav version 6.0.1
libdvdread: Encrypted DVD support unavailable.
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.IFO failed
libdvdread:DVDOpenFileUDF:UDFFindFile /VIDEO_TS/VIDEO_TS.BUP failed
libdvdread: Can't open file VIDEO_TS.IFO.
libdvdnav: vm: failed to read VIDEO_TS.IFO
[09:39:38] dvd: not a dvd - trying as a stream/file instead
[09:39:38] file is MPEG Program Stream
[09:39:38] Probing 1 unknown stream
[09:39:38]     Probe: Found stream mpegvideo. stream id 0xe0-0x0
[09:39:38] Found the following streams
[09:39:38]     Video Streams :
[09:39:38]       0xe0-0x0 type mpegvideo (0x2)
[09:39:38]     Audio Streams :
[09:39:38]       0xbd-0x80 type AC3 (0x81)
[09:39:38]     Subtitle Streams :
[09:39:38]     Other Streams :
[09:39:38] stream id 0xbd (type 0x81 substream 0x80) audio 0x8000bd
[09:39:38] hb_sample_pts: couldn't find video packet near 7081984
[09:39:38] hb_sample_pts: couldn't find video packet near 7139328
[09:39:38] hb_sample_pts: couldn't find video packet near 7198720
[09:39:38] hb_sample_pts: couldn't find video packet near 7256064
[09:39:38] hb_sample_pts: couldn't find video packet near 7313408
[09:39:38] hb_sample_pts: couldn't find video packet near 7370752
[09:39:38] scan: decoding previews for title 1
[09:39:38] file is MPEG Program Stream
[09:39:38] Probing 1 unknown stream
[09:39:38]     Probe: Found stream mpegvideo. stream id 0xe0-0x0
[09:39:38] scan: preview 1
[09:39:38] fifo_close: trashing 0 buffer(s)
[09:39:38] scan: audio 0x8000bd: ac3, rate=48000Hz, bitrate=448000 Unknown (AC3) (5.1 ch) (448 kbps)
[09:39:38] scan: preview 2
[09:39:38] scan: preview 3
[09:39:38] scan: preview 4
[09:39:38] scan: preview 5
[09:39:38] scan: preview 6
[09:39:38] scan: preview 7
[09:39:38] scan: preview 8
[09:39:38] scan: preview 9
[09:39:38] scan: preview 10
[09:39:38] stream: 73 good frames, 0 errors (0%)
[09:39:38] scan: 10 previews, 720x576, 25.000 fps, autocrop = 2/52/0/118, aspect 16:9, PAR 64:45, color profile: 5-1-6
[09:39:38] Freed 1 buffers of size 4096
[09:39:38] Freed 2 buffers of size 65536
[09:39:38] Freed 9 buffers of size 1048576
[09:39:38] Allocated 9572352 bytes of buffers on this pass and Freed 9572352 bytes, 0 bytes leaked
[09:39:38] thread 2e exited ("scan")
[09:39:38] thread 2e joined ("scan")
[09:39:38] libhb: scan thread found 1 valid title(s)
[09:39:39] Starting Task: Encoding Pass
[09:39:39] Skipping vfr filter
[09:39:39] thread 2f started ("decomb_filter_segment")
[09:39:39] thread 30 started ("decomb_filter_segment")
[09:39:39] thread 31 started ("decomb_filter_segment")
[09:39:39] thread 32 started ("decomb_filter_segment")
[09:39:39] thread 33 started ("decomb_check_segment")
[09:39:39] thread 34 started ("decomb_check_segment")
[09:39:39] thread 35 started ("decomb_check_segment")
[09:39:39] thread 36 started ("decomb_check_segment")
[09:39:39] thread 37 started ("mask_filter_segment")
[09:39:39] thread 38 started ("mask_filter_segment")
[09:39:39] thread 39 started ("mask_filter_segment")
[09:39:39] thread 3a started ("mask_filter_segment")
[09:39:39] thread 3b started ("mask_erode_segment")
[09:39:39] thread 3c started ("mask_erode_segment")
[09:39:39] thread 3d started ("mask_erode_segment")
[09:39:39] thread 3e started ("mask_erode_segment")
[09:39:39] thread 3f started ("mask_dilate_segment")
[09:39:39] thread 40 started ("mask_dilate_segment")
[09:39:39] thread 41 started ("mask_dilate_segment")
[09:39:39] thread 42 started ("mask_dilate_segment")
[09:39:39] thread 43 started ("yadif_filter_segment")
[09:39:39] thread 44 started ("yadif_filter_segment")
[09:39:39] thread 45 started ("yadif_filter_segment")
[09:39:39] thread 46 started ("yadif_filter_segment")
[09:39:39] work: track 1, dithering not supported by codec
[09:39:39] work: only 1 chapter, disabling chapter markers
[09:39:39] job configuration:
[09:39:39]  * source
[09:39:39]    + V:\0-9\!!!\S01\distributor_walt_disney_enhanced-DWEU.vob
[09:39:39]    + title 1, start 00:00:0.00 stop 00:00:30.00
[09:39:39]  * destination
[09:39:39]    + C:\Users\Media Server\Videos\Distributor Walt Disney Enhanced-Dweu-1_preview.m4v
[09:39:39]    + container: MPEG-4 (libavformat)
[09:39:39]      + align initial A/V stream timestamps
[09:39:39]  * video track
[09:39:39]    + decoder: mpeg2video 8-bit
[09:39:39]      + bitrate 200 kbps
[09:39:39]    + filters
[09:39:39]      + Comb Detect (mode=3:spatial-metric=2:motion-thresh=1:spatial-thresh=1:filter-mode=2:block-thresh=40:block-width=16:block-height=16)
[09:39:39]      + Decomb (mode=39)
[09:39:39]      + AVFilter (crop='x=0:y=2:w=602:h=522',scale='width=602:height=522:flags=lanczos+accurate_rnd:in_color_matrix=smpte170m:out_color_matrix=smpte170m:out_range=limited',format='pix_fmts=yuv420p')
[09:39:39]        + crop: x=0, y=2, w=602, h=522
[09:39:39]        + scale: width=602, height=522, flags=lanczos+accurate_rnd, in_color_matrix=smpte170m, out_color_matrix=smpte170m, out_range=limited
[09:39:39]        + format: pix_fmts=yuv420p
[09:39:39]      + Crop and Scale (width=602:height=522:crop-top=2:crop-bottom=52:crop-left=0:crop-right=118)
[09:39:39]        + source: 720 * 576, crop (2/52/0/118): 602 * 522, scale: 602 * 522
[09:39:39]    + Output geometry
[09:39:39]      + storage dimensions: 602 x 522
[09:39:39]      + pixel aspect ratio: 64 : 45
[09:39:39]      + display dimensions: 856 x 522
[09:39:39]    + encoder: H.265 (Intel Media SDK)
[09:39:39]      + preset:  balanced
[09:39:39]      + options: lowpower=1
[09:39:39]      + profile: auto
[09:39:39]      + level:   auto
[09:39:39]      + quality: 22.00 (QP)
[09:39:39]      + color profile: 5-1-6
[09:39:39]  * audio track 1
[09:39:39]    + decoder: Unknown (AC3) (5.1 ch) (448 kbps) (track 1, id 0x8000bd)
[09:39:39]      + bitrate: 448 kbps, samplerate: 48000 Hz
[09:39:39]    + mixdown: Stereo
[09:39:39]    + encoder: AAC (libavcodec)
[09:39:39]      + bitrate: 160 kbps, samplerate: 48000 Hz
[09:39:39] file is MPEG Program Stream
[09:39:39] Probing 1 unknown stream
[09:39:39]     Probe: Found stream mpegvideo. stream id 0xe0-0x0
[09:39:39] sync: expecting 775 video frames
[09:39:39] thread 47 started ("Audio Synchronization")
qsv: incorrect qsv device index -1
Last error repeated 2 times
hb_qsv_param_default_preset: invalid pointer(s) param=0000021d734f4fd8 videoParam=0000021d734f5100 info=0000000000000000 preset=0000021d7325c630
encqsvInit: hb_qsv_param_default_preset failed
Failure to initialise thread 'Quick Sync Video encoder (Intel Media SDK)'
[09:39:39] comb detect: heavy 0 | light 0 | uncombed 0 | total 0
[09:39:39] thread 32 exited ("decomb_filter_segment")
[09:39:39] thread 30 exited ("decomb_filter_segment")
[09:39:39] thread 31 exited ("decomb_filter_segment")
[09:39:39] thread 2f exited ("decomb_filter_segment")
[09:39:39] thread 2f joined ("decomb_filter_segment")
[09:39:39] thread 30 joined ("decomb_filter_segment")
[09:39:39] thread 31 joined ("decomb_filter_segment")
[09:39:39] thread 32 joined ("decomb_filter_segment")
[09:39:39] thread 33 exited ("decomb_check_segment")
[09:39:39] thread 36 exited ("decomb_check_segment")
[09:39:39] thread 34 exited ("decomb_check_segment")
[09:39:39] thread 35 exited ("decomb_check_segment")
[09:39:39] thread 33 joined ("decomb_check_segment")
[09:39:39] thread 34 joined ("decomb_check_segment")
[09:39:39] thread 35 joined ("decomb_check_segment")
[09:39:39] thread 36 joined ("decomb_check_segment")
[09:39:39] thread 3a exited ("mask_filter_segment")
[09:39:39] thread 38 exited ("mask_filter_segment")
[09:39:39] thread 39 exited ("mask_filter_segment")
[09:39:39] thread 37 exited ("mask_filter_segment")
[09:39:39] thread 37 joined ("mask_filter_segment")
[09:39:39] thread 38 joined ("mask_filter_segment")
[09:39:39] thread 39 joined ("mask_filter_segment")
[09:39:39] thread 3a joined ("mask_filter_segment")
[09:39:39] thread 3b exited ("mask_erode_segment")
[09:39:39] thread 3e exited ("mask_erode_segment")
[09:39:39] thread 3c exited ("mask_erode_segment")
[09:39:39] thread 3d exited ("mask_erode_segment")
[09:39:39] thread 3b joined ("mask_erode_segment")
[09:39:39] thread 3c joined ("mask_erode_segment")
[09:39:39] thread 3d joined ("mask_erode_segment")
[09:39:39] thread 3e joined ("mask_erode_segment")
[09:39:39] thread 3f exited ("mask_dilate_segment")
[09:39:39] thread 41 exited ("mask_dilate_segment")
[09:39:39] thread 42 exited ("mask_dilate_segment")
[09:39:39] thread 40 exited ("mask_dilate_segment")
[09:39:39] thread 3f joined ("mask_dilate_segment")
[09:39:39] thread 40 joined ("mask_dilate_segment")
[09:39:39] thread 41 joined ("mask_dilate_segment")
[09:39:39] thread 42 joined ("mask_dilate_segment")
[09:39:39] decomb: deinterlaced 0 | blended 0 | unfiltered 0 | total 0
[09:39:39] thread 45 exited ("yadif_filter_segment")
[09:39:39] thread 46 exited ("yadif_filter_segment")
[09:39:39] thread 44 exited ("yadif_filter_segment")
[09:39:39] thread 43 exited ("yadif_filter_segment")
[09:39:39] thread 43 joined ("yadif_filter_segment")
[09:39:39] thread 44 joined ("yadif_filter_segment")
[09:39:39] thread 45 joined ("yadif_filter_segment")
[09:39:39] thread 46 joined ("yadif_filter_segment")
[09:39:39] ac3-decoder done: 0 frames, 0 decoder errors
[09:39:39] mpeg2video-decoder done: 0 frames, 0 decoder errors
[09:39:39] sync: got 0 frames, 775 expected
[09:39:39] thread 47 exited ("Audio Synchronization")
[09:39:39] thread 47 joined ("Audio Synchronization")
[09:39:39] encavcodecaudio: closing libavcodec
[aac @ 0000021d7e6bbac0] Ignoring attempt to flush encoder that doesn't support it
[aac @ 0000021d7e6bbac0] Qavg: nan
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] fifo_close: trashing 0 buffer(s)
[09:39:39] Freed 1 buffers of size 4096
[09:39:39] Freed 1 buffers of size 65536
[09:39:39] Freed 3 buffers of size 1048576
[09:39:39] Allocated 3215360 bytes of buffers on this pass and Freed 3215360 bytes, 0 bytes leaked
[09:39:39] Finished work at: Sun Mar 21 09:39:39 2021
[09:39:39] thread 2d exited ("work")
[09:39:39] thread 2d joined ("work")
[09:39:39] libhb: work result = 3

# Job Failed to Initialise. Check log and input settings (3)

```

--- a/libhb/common.c
+++ b/libhb/common.c
@@ -3899,7 +3899,6 @@ static void job_setup(hb_job_t * job, hb_title_t * title)
 #if HB_PROJECT_FEATURE_QSV
     job->qsv.ctx                   = hb_qsv_context_init();
     job->qsv.enc_info.is_init_done = 0;
-    job->qsv.async_depth           = hb_qsv_param_default_async_depth();
     job->qsv.decode                = !!(title->video_decode_support &
                                         HB_DECODE_SUPPORT_QSV);
 #endif

--- a/libhb/decavcodec.c
+++ b/libhb/decavcodec.c
@@ -2132,8 +2132,11 @@ static int decavcodecvInfo( hb_work_object_t *w, hb_work_info_t *info )
     info->video_decode_support = HB_DECODE_SUPPORT_SW;
 
 #if HB_PROJECT_FEATURE_QSV
-    if (hb_qsv_decode_codec_supported_codec(hb_qsv_get_adapter_index(), pv->context->codec_id, pv->context->pix_fmt))
-        info->video_decode_support |= HB_DECODE_SUPPORT_QSV;
+    if (hb_qsv_available())
+    {
+        if (hb_qsv_decode_codec_supported_codec(hb_qsv_get_adapter_index(), pv->context->codec_id, pv->context->pix_fmt))
+            info->video_decode_support |= HB_DECODE_SUPPORT_QSV;
+    }
 #endif
 
     return 1;

--- a/libhb/qsv_common.c
+++ b/libhb/qsv_common.c
@@ -56,7 +56,8 @@ static const char* hb_qsv_get_adapter_type(const mfxAdapterInfo* info);
 int hb_qsv_get_platform(int adapter_index);
 static hb_list_t *g_qsv_adapters_list         = NULL;
 static hb_list_t *g_qsv_adapters_details_list = NULL;
-static int g_adapter_index = -1;
+static int g_adapter_index = 0;
+static int g_default_adapter_index = 0;
 
 static void init_adapter_details(hb_qsv_adapter_details_t *adapter_details)
 {
@@ -166,6 +167,17 @@ static hb_triplet_t hb_qsv_h265_levels[] =
 int hb_qsv_get_adapter_index()
 {
     return g_adapter_index;
+}
+
+static int hb_qsv_set_default_adapter_index(int adapter_index)
+{
+    g_default_adapter_index = adapter_index;
+    return 0;
+}
+
+static int hb_qsv_get_default_adapter_index()
+{
+    return g_default_adapter_index;
 }
 
 hb_list_t* hb_qsv_adapters_list()
@@ -933,7 +945,7 @@ static int hb_qsv_collect_adapters_details(hb_list_t *qsv_adapters_list, hb_list
     return 0;
 }
 
-static void log_decoder_capabilities(int log_level, hb_qsv_adapter_details_t *adapter_details, const char *prefix)
+static void log_decoder_capabilities(const int log_level, const hb_qsv_adapter_details_t *adapter_details, const char *prefix)
 {
     char buffer[128] = "";
 
@@ -960,7 +972,7 @@ static void log_decoder_capabilities(int log_level, hb_qsv_adapter_details_t *ad
                 strnlen(buffer, 1) ? buffer : " no decode support");
 }
 
-static void log_encoder_capabilities(int log_level, uint64_t caps, const char *prefix)
+static void log_encoder_capabilities(const int log_level, const uint64_t caps, const char *prefix)
 {
     /*
      * Note: keep the string short, as it may be logged by default.
@@ -1046,7 +1058,7 @@ static void log_encoder_capabilities(int log_level, uint64_t caps, const char *p
                 strnlen(buffer, 1) ? buffer : " standard feature set");
 }
 
-static void hb_qsv_adapter_info_print(hb_qsv_adapter_details_t *adapter_details)
+static void hb_qsv_adapter_info_print(const hb_qsv_adapter_details_t *adapter_details)
 {
     if (adapter_details->qsv_hardware_version.Version)
     {
@@ -1136,7 +1148,7 @@ void hb_qsv_info_print()
         // also print the details about all QSV adapters
         for (int i = 0; i < hb_list_count(g_qsv_adapters_details_list); i++)
         {
-            hb_qsv_adapter_details_t *details = hb_list_item(g_qsv_adapters_details_list, i);
+            const hb_qsv_adapter_details_t *details = hb_list_item(g_qsv_adapters_details_list, i);
 #if defined(_WIN32) || defined(__MINGW32__)
             mfxAdapterInfo* info = &g_qsv_adapters_info.Adapters[i];
             hb_log("Intel Quick Sync Video %s adapter with index %d", hb_qsv_get_adapter_type(info), details->index);
@@ -1310,7 +1322,7 @@ int hb_qsv_decode_codec_supported_codec(int adapter_index, int video_codec_param
 int hb_qsv_setup_job(hb_job_t *job)
 {
 #if defined(_WIN32) || defined(__MINGW32__)
-    if (job->qsv.ctx->dx_index >= 0)
+    if (job->qsv.ctx && job->qsv.ctx->dx_index >= 0)
     {
         hb_qsv_param_parse_dx_index(job, job->qsv.ctx->dx_index);
     }
@@ -1365,7 +1377,7 @@ int hb_qsv_full_path_is_enabled(hb_job_t *job)
 
     qsv_full_path_is_enabled = (hb_qsv_decode_is_enabled(job) &&
         info && hb_qsv_implementation_is_hardware(info->implementation) &&
-        device_check_succeded && !job->qsv.ctx->num_cpu_filters) && !codecs_exceptions;
+        device_check_succeded && job->qsv.ctx && !job->qsv.ctx->num_cpu_filters) && !codecs_exceptions;
     return qsv_full_path_is_enabled;
 }
 
@@ -1981,7 +1993,7 @@ int hb_qsv_param_parse(hb_qsv_param_t *param, hb_qsv_info_t *info, hb_job_t *job
     else if (!strcasecmp(key, "gpu"))
     {
         // Check if was parsed already in decoder initialization
-        if (!job->qsv.ctx->qsv_device)
+        if (job->qsv.ctx && !job->qsv.ctx->qsv_device)
         {
             int gpu_index = hb_qsv_atoi(value, &error);
             if (!error)
@@ -2735,6 +2747,9 @@ int hb_qsv_info_init()
 
 int hb_qsv_set_adapter_index(int adapter_index)
 {
+    if (g_adapter_index == adapter_index)
+        return 0;
+
     for (int i = 0; i < g_qsv_adapters_info.NumActual; i++)
     {
         mfxAdapterInfo* info = &g_qsv_adapters_info.Adapters[i];
@@ -2840,7 +2855,7 @@ int hb_qsv_get_platform(int adapter_index)
             return qsv_map_mfx_platform_codename(info->Platform.CodeName);
         }
     }
-    hb_error("qsv: incorrect qsv device index %d", adapter_index);
+    hb_error("qsv: hb_qsv_get_platform incorrect qsv device index %d", adapter_index);
     return HB_CPU_PLATFORM_UNSPECIFIED;
 }
 
@@ -2853,7 +2868,7 @@ int hb_qsv_param_parse_dx_index(hb_job_t *job, const int dx_index)
         // if -1 use first adapter with highest priority
         if (info && ((info->Number == dx_index) || (dx_index == -1)))
         {
-            if (!job->qsv.ctx->qsv_device)
+            if (job->qsv.ctx && !job->qsv.ctx->qsv_device)
             {
                 job->qsv.ctx->qsv_device = av_mallocz_array(32, sizeof(*job->qsv.ctx->qsv_device));
                 if (!job->qsv.ctx->qsv_device)
@@ -2869,7 +2884,7 @@ int hb_qsv_param_parse_dx_index(hb_job_t *job, const int dx_index)
             return 0;
         }
     }
-    hb_error("qsv: incorrect qsv device index %d", dx_index);
+    hb_error("qsv: hb_qsv_param_parse_dx_index incorrect qsv device index %d", dx_index);
     return -1;
 }
 
@@ -3318,6 +3333,7 @@ static int hb_qsv_get_dx_device(hb_job_t *job)
 static int hb_qsv_make_adapters_list(const mfxAdaptersInfo* adapters_info, hb_list_t **qsv_adapters_list)
 {
     int max_generation = QSV_G0;
+    int default_adapter = 0;
 
     if (!qsv_adapters_list)
     {
@@ -3345,11 +3361,13 @@ static int hb_qsv_make_adapters_list(const mfxAdaptersInfo* adapters_info, hb_li
             if (generation > max_generation || info->Platform.MediaAdapterType == MFX_MEDIA_DISCRETE)
             {
                 max_generation = generation;
-                hb_qsv_set_adapter_index(info->Number);
+                default_adapter = info->Number;
             }
             hb_list_add(list, (void*)&info->Number);
         }
     }
+    hb_qsv_set_default_adapter_index(default_adapter);
+    hb_qsv_set_adapter_index(default_adapter);
     *qsv_adapters_list = list;
     return 0;
 }
@@ -3597,14 +3615,14 @@ void hb_qsv_uninit_dec(AVCodecContext *s)
 
 void hb_qsv_uninit_enc(hb_job_t *job)
 {
-    if(job->qsv.ctx->hb_dec_qsv_frames_ctx)
+    if(job->qsv.ctx && job->qsv.ctx->hb_dec_qsv_frames_ctx)
     {
         av_buffer_unref(&job->qsv.ctx->hb_dec_qsv_frames_ctx->hw_frames_ctx);
         job->qsv.ctx->hb_dec_qsv_frames_ctx->hw_frames_ctx = NULL;
         av_free(job->qsv.ctx->hb_dec_qsv_frames_ctx);
         job->qsv.ctx->hb_dec_qsv_frames_ctx = NULL;
     }
-    if(job->qsv.ctx->hb_vpp_qsv_frames_ctx)
+    if(job->qsv.ctx && job->qsv.ctx->hb_vpp_qsv_frames_ctx)
     {
         av_buffer_unref(&job->qsv.ctx->hb_vpp_qsv_frames_ctx->hw_frames_ctx);
         job->qsv.ctx->hb_vpp_qsv_frames_ctx->hw_frames_ctx = NULL;
@@ -3616,11 +3634,14 @@ void hb_qsv_uninit_enc(hb_job_t *job)
         ID3D11DeviceContext_Release(device_context);
         device_context = NULL;
     }
-    job->qsv.ctx->hb_hw_device_ctx = NULL;
-    if (job->qsv.ctx->qsv_device)
+    if (job->qsv.ctx)
     {
-        av_free(job->qsv.ctx->qsv_device);
-        job->qsv.ctx->qsv_device = NULL;
+        job->qsv.ctx->hb_hw_device_ctx = NULL;
+        if (job->qsv.ctx->qsv_device)
+        {
+            av_free(job->qsv.ctx->qsv_device);
+            job->qsv.ctx->qsv_device = NULL;
+        }
     }
     device_manager_handle = NULL;
 }
@@ -3630,7 +3651,7 @@ static int qsv_device_init(hb_job_t *job)
     int err;
     AVDictionary *dict = NULL;
 
-    if (job->qsv.ctx->qsv_device)
+    if (job->qsv.ctx && job->qsv.ctx->qsv_device)
     {
         err = av_dict_set(&dict, "child_device", job->qsv.ctx->qsv_device, 0);
         if (err < 0)
@@ -3696,7 +3717,7 @@ int hb_create_ffmpeg_pool(hb_job_t *job, int coded_width, int coded_height, enum
 
     int ret = 0;
 
-    if (!job->qsv.ctx->hb_hw_device_ctx) {
+    if (job->qsv.ctx && !job->qsv.ctx->hb_hw_device_ctx) {
         // parse and use user-specified encoder options for decoder, if present
         if (job->encoder_options != NULL && *job->encoder_options)
         {
@@ -3988,7 +4009,7 @@ int hb_qsv_info_init()
         hb_error("hb_qsv_info_init: g_qsv_adapters_details_list allocation failed");
         return -1;
     }
-    static int adapter_index = -1;
+    static int adapter_index = 0;
     static hb_qsv_adapter_details_t adapter_details;
     init_adapter_details(&adapter_details);
     hb_list_add(g_qsv_adapters_list, (void*)&adapter_index);
@@ -4097,6 +4118,10 @@ void hb_qsv_context_uninit(hb_job_t *job)
     av_free(ctx);
     job->qsv.ctx = NULL;
 
+    /* Structures below are needed until the end life of the process
+        It has been collected once in hb_qsv_info_init() and no need to recollect every time.
+    */
+#if 0
     if (g_qsv_adapters_details_list)
     {
 #if defined(_WIN32) || defined(__MINGW32__)
@@ -4119,7 +4144,8 @@ void hb_qsv_context_uninit(hb_job_t *job)
     g_qsv_adapters_info.NumAlloc = 0;
     g_qsv_adapters_info.NumActual = 0;
 #endif
-    g_adapter_index = -1;
+#endif
+    g_adapter_index = hb_qsv_get_default_adapter_index();
 }
 
 #else // HB_PROJECT_FEATURE_QSV

--- a/libhb/work.c
+++ b/libhb/work.c
@@ -142,7 +142,10 @@ static void work_func( void * _work )
             job = new_job;
         }
 #if HB_PROJECT_FEATURE_QSV
-        hb_qsv_setup_job(job);
+        if (hb_qsv_available())
+        {
+            hb_qsv_setup_job(job);
+        }
 #endif
         hb_job_setup_passes(job->h, job, passes);
         hb_job_close(&job);

